kratos.gid/apps/Structural/xml/ConstitutiveLaws.xml

```diff
@@ -78,23 +78,37 @@
             <parameter n="DENSITY" pn="Density" unit_magnitude="Density" units="kg/m^3" v="7850"/>
             <parameter n="YOUNG_MODULUS" pn="Young Modulus" unit_magnitude="P" units="Pa" v="206.9e9" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.CROSS_AREA" pn="Cross area" v="1.0" unit_magnitude="Area" units="m^2" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.TRUSS_PRESTRESS_PK2" pn="Prestress" type="double" v="0.0" unit_magnitude="P" units="Pa" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.TRUSS_IS_CABLE" pn="Is cable" type="bool" v="false" />
+            <parameter n="CROSS_AREA" pn="Cross area" v="1.0" unit_magnitude="Area" units="m^2" />
+            <parameter n="TRUSS_PRESTRESS_PK2" pn="Prestress" type="double" v="0.0" unit_magnitude="P" units="Pa" />
         </inputs>
         <outputs></outputs>
     </CLaw>
 
     <!--constitutive laws for beams-->
-    <CLaw n="BeamConstitutiveLaw" pn="Linear elastic beam" ProductionReady="ProductionReady" Type="Beam" behaviour="Elastic" StrainSize="1" ImplementedInApplication="StructuralMechanicsApplication" help="Linear elastic behaviour for beams" Dimension="2D,3D" LargeDeformation="False" RequiresLocalAxes="False" HybridType="False">
+    <CLaw n="BeamConstitutiveLaw" pn="Linear elastic beam" ProductionReady="ProductionReady" Type="Beam2D" behaviour="Elastic"
+          StrainSize="1" ImplementedInApplication="StructuralMechanicsApplication" help="Linear elastic behaviour for beams"
+          Dimension="2D" LargeDeformation="False" RequiresLocalAxes="False" HybridType="False">
         <inputs>
             <parameter n="DENSITY" pn="Density" unit_magnitude="Density" units="kg/m^3" v="7850"/>
             <parameter n="YOUNG_MODULUS" pn="Young Modulus" unit_magnitude="P" units="Pa" v="206.9e9" />
             <parameter n="POISSON_RATIO" pn="Poisson Ratio" v="0.29"/>
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.CROSS_AREA" pn="Cross area" v="1.0" unit_magnitude="Area" units="m^2" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.TORSIONAL_INERTIA" pn="Torsional inertia" type="double" v="1.0" unit_magnitude="L^4" units="m^4" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.I22" pn="Inertia 22" type="double" v="1.0" unit_magnitude="L^4" units="m^4" />
-            <parameter n="KratosMultiphysics.StructuralMechanicsApplication.I33" pn="Inertia 33" type="double" v="1.0" unit_magnitude="L^4" units="m^4" />
+            <parameter n="CROSS_AREA" pn="Cross area" v="1.0" unit_magnitude="Area" units="m^2" />
+            <parameter n="I33" pn="Inertia 33" type="double" v="1.0" unit_magnitude="L^4" units="m^4" />
+        </inputs>
+        <outputs></outputs>
+    </CLaw>
+
+    <CLaw n="BeamConstitutiveLaw" pn="Linear elastic beam" ProductionReady="ProductionReady" Type="Beam3D" behaviour="Elastic"

```

Comment:
@philbucher The 'n' field is the Primary Key for the constitutive law database, so 2 CLaw can not have the same n

In this case, I see 2 CLaw with  n="BeamConstitutiveLaw"

Reply:
Yes because the name is the same for 2D and 3D but the variables to be written are different
I checked the elements, which filter the ConstLaw by `Type`, wich I changed to `Beam2D` and `Beam3D` respectively

Reply:
Well, they must have different names if they're different. The only use case for using twice the same name is to replace the previous one.

Reply:
Or, we could think about a tricky solution, if they must be named with the same way

Reply:
I don't think it's worth it, I will revert the change

Reply:
I just saw that `LinearElasticPlaneStress2DLaw` exists three times with the same name, how does this work?
Isn't this the same I am trying to do?

Reply:
Nice discovery, It is not working, properly, and it's almost impossible to debug.

I was working on a local branch to add a "Kratos_Name" tag to the constitutive laws, so the 'n' becomes into an identifier only relevant to the GUI, and the printed name can be repeated.

Let me try to merge it before this pull request and let's check if it works